src/WebJobs.Extensions.Twilio/Config/TwilioSMSConfiguration.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Twilio
+{
+    /// <summary>
+    /// Defines the configuration options for the Twilio binding.
+    /// </summary>
+    public class TwilioSmsConfiguration
+    {
+        internal const string AzureWebJobsTwilioAccountSidKeyName = "AzureWebJobsTwilioAccountSid";
+        internal const string AzureWebJobsTwilioAccountAuthTokenName = "AzureWebJobsTwilioAuthToken";
+
+        public TwilioSmsConfiguration()
+        {
+            var nameResolver = new DefaultNameResolver();
+            AccountSid = nameResolver.Resolve(AzureWebJobsTwilioAccountSidKeyName);
```

Comment:
That nameresolver is used for [AutoResolve] -- which is correct. But here you're using it to set the default values. If someone were to plug in a TestNameResolver, it wouldn't get used here. 

The 'Attribute wins' approach (that SendGrid does not use) made this more complicated. I'd check out something like NH (which is probably the simplest). 

---
In reply to: [75050475](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/124#discussion_r75050475) [](ancestors = 75050475)